gordon-plugin/src/main/kotlin/com/banno/gordon/GordonTestTask.kt

```diff
@@ -197,7 +202,7 @@ internal abstract class GordonTestTask @Inject constructor(
 
             val htmlReportPath = testResults.htmlReport().write(reportDirectory.get().asFile).bind()
 
-            pools.flatMap { it.devices }.safeUninstall(
+            pools.filterProblematicDevices(problematicDevices).flatMap { it.devices }.safeUninstall(
```

Comment:
`safeUninstall` already ignores failures, so we might as well try to do as much cleanup as possible and try it on all devices

```suggestion
            originalPools.flatMap { it.devices }.safeUninstall(
```